src/providers/twitch/TwitchChannel.cpp

```diff
@@ -104,6 +105,14 @@ TwitchChannel::TwitchChannel(const QString &name)
         this->loadRecentMessagesReconnect();
     });
 
+    this->destroyed.connect([this]() {
+        if (getApp()->twitch->seventvEventAPI)
```

Comment:
No need to check if `seventvEventAPI` is set if it's already done in the follow-up functions

and `13:10:03 nerixyz: these two functions could be meged now that I think about it `